packages/components/src/interfaces/internal/GoogleTagManager.ts

```diff
@@ -0,0 +1,21 @@
+import type { ScriptComponentType } from "~/types"
+
+export interface GoogleTagManagerHeaderScriptProps {
+  gtmId: string
+  ScriptComponent?: ScriptComponentType
+}
+
+export interface GoogleTagManagerHeaderProps {
+  siteGtmId: string | undefined
+  isomerGtmId: string | undefined
+  ScriptComponent?: ScriptComponentType
+}
```

Comment:
nit: I think we can define the props once then build on top of it for these Header props?

```ts
export interface GoogleTagManagerBodyScriptProps {
  gtmId: string
}

export interface GoogleTagManagerHeaderScriptProps extends GoogleTagManagerBodyScriptProps {
  ScriptComponent?: ScriptComponentType
}
```

This way, we can define the type for the gtmId once. Same thing for the `GoogleTagManagerBodyProps`.

Reply:
what's the best way to do this since we are using different variable name e.g. `gtmId` and `siteGtmId`?